deploy/jbossas/modeshape-jbossas-subsystem/src/main/java/org/modeshape/jboss/subsystem/AddCustomBinaryStorage.java

```diff
@@ -39,17 +45,28 @@ protected void writeBinaryStorageConfiguration( String repositoryName,
                                                     ModelNode model,
                                                     EditableDocument binaries ) {
         binaries.set(FieldName.TYPE, FieldValue.BINARY_STORAGE_TYPE_CUSTOM);
-        for (Property property : model.asPropertyList()) {
-            String name = property.getName();
-            if (name.equals(ModelKeys.CLASSNAME)) {
-                name = FieldName.CLASSNAME;
-            } else if (name.equals(ModelKeys.MODULE)) {
-                name = FieldName.CLASSLOADER;
-            }
-            binaries.set(name, property.getValue());
+        model.asPropertyList().forEach(property -> writeCustomProperty(binaries, property));
+    }
+    
+    private void writeCustomProperty(EditableDocument binaries, Property property) {
+        ModelNode value = property.getValue();
+        if (!value.isDefined()) {
+            return;
         }
+        String name = property.getName();
+        if (COMMON_BINARY_ATTRIBUTES.contains(name)) {
+            // these are set via the base class...
+            return;
```

Comment:
How does this work to set them via the base class? This method doesn't call 'super'.